jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -357,7 +358,7 @@ static String toRedirectURI(Request request, String location)
             {
                 // make the location an absolute URI
                 StringBuilder url = new StringBuilder(128);
-                URIUtil.appendSchemeHostPort(url, uri.getScheme(), Request.getServerName(request), Request.getServerPort(request));
+                HttpScheme.appendNormalizedUri(url, uri.getScheme(), Request.getServerName(request), Request.getServerPort(request));
```

Comment:
You can use the method that also takes the path and query, since you are adding the location in the next line.

Reply:
Nope, as `location` already has the query considered, it is not a pure path.